src/Objects/_PVector.js

```diff
@@ -0,0 +1,283 @@
+module.exports = function(options, undef) {
+  'use strict';
+
+  const TAU = 2 * Math.PI,
+        //TAU = options.PConstants.TAU,
+        lerp = function(start, stop, amt) {
+          return amt * (stop - start) + start;
+        },
+        sq = function(n) {
+          return n*n;
+        },
+        argsErr = function(mtd, len, min) {
+          throw 'Too few args passed to ' + mtd +
+                '() [' + len + ' < ' + min + '].';
+        };
+
+  function _PVector(x, y, z) {
+    this.x = +x || 0;
+    this.y = +y || 0;
+    this.z = +z || 0;
+  }
+
+  _PVector.fromAngle = function(angle, target) {
+    return target? target.set(Math.cos(angle), Math.sin(angle))
+                 : new _PVector(Math.cos(angle), Math.sin(angle));
+  };
+
+  _PVector.random2D = function(target, parent) {
+    //const rnd = parent? parent : target instanceof Processing? target : Math;
+    const rnd = parent? parent : target && target.random? target : Math;
+    return _PVector.fromAngle(TAU * rnd.random(), target);
+  };
+
+  _PVector.random3D = function(target, parent) {
+    //const rnd = parent? parent : target instanceof Processing? target : Math,
+    const rnd = parent? parent : target && target.random? target : Math,
+          ang = TAU * rnd.random(),
+          vz  = 2 * rnd.random() - 1,
+          vzr = Math.sqrt(1 - vz*vz),
+          vx  = vzr * Math.cos(ang),
+          vy  = vzr * Math.sin(ang);
+    return target? target.set(vx, vy, vz) : new _PVector(vx, vy, vz);
+  };
+
+  _PVector.dist = function(v1, v2) {
+    return Math.sqrt(_PVector.distSq(v1, v2));
+  };
+
+  _PVector.distSq = function(v1, v2) {
+    return sq(v1.x - v2.x) + sq(v1.y - v2.y) + sq(v1.z - v2.z);
+  };
+
+  _PVector.dot = function(v1, v2) {
+    return v1.x*v2.x + v1.y*v2.y + v1.z*v2.z;
+  };
+
+  _PVector.cross = function(v1, v2, target) {
```

Comment:
It looks like this code needs a branch to check whether we're dealing with a 2D or 3D vector, since the cross product in 2D is not the same as a cross product in 3D with zero `z` component.